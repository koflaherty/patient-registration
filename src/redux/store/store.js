import { createStore, combineReducers } from 'redux';
import { reducer as reduxFormReducer } from 'redux-form';
import { enteredRegistration } from "../reducers/reducers";

export const ENTERED_REGISTRATION_STATE_KEY = 'registration';

const reducer = combineReducers({
    form: reduxFormReducer,
    [ENTERED_REGISTRATION_STATE_KEY]: enteredRegistration,
});

export const store = (window.devToolsExtension
    ? window.devToolsExtension()(createStore)
    : createStore)(reducer);
