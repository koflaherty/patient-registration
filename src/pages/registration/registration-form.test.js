import React from 'react';
import RegistrationForm from './registration-form';

import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import {store} from "../../redux/store/store";

it('renders correctly', () => {
    const tree = renderer
        .create(<Provider store={store}><RegistrationForm/></Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});