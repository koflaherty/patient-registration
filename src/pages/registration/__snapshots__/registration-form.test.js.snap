// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<form
  className="ant-form ant-form-vertical"
  onSubmit={[Function]}
>
  <h2>
    Patient Information
  </h2>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="First Name"
      >
        First Name
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            required={true}
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Last Name"
      >
        Last Name
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            required={true}
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Date of Birth"
      >
        Date of Birth
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <span
            className="ant-calendar-picker"
            onBlur={[Function]}
            onFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={Object {}}
          >
            <div
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
            >
              <input
                className="ant-calendar-picker-input ant-input"
                placeholder="Select date"
                readOnly={true}
                value=""
              />
              <i
                className="anticon anticon-calendar ant-calendar-picker-icon"
              >
                <svg
                  aria-hidden="true"
                  className=""
                  data-icon="calendar"
                  fill="currentColor"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                  />
                </svg>
              </i>
            </div>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Email (optional)"
      >
        Email (optional)
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="email"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Home Address"
      >
        Home Address
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <input
            className="ant-input"
            disabled={false}
            name="homeAddress"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            required={true}
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <h2>
    Medical History (optional)
  </h2>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Family History"
      >
        Family History
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <textarea
            className="ant-input"
            name="familyHistory"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Medications"
      >
        Medications
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <textarea
            className="ant-input"
            name="medications"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Diseases"
      >
        Diseases
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <textarea
            className="ant-input"
            name="diseases"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="Allergies"
      >
        Allergies
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <textarea
            className="ant-input"
            name="allergies"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={Object {}}
            value=""
          />
        </span>
      </div>
    </div>
  </div>
  <h2>
    Terms and Conditions
  </h2>
  <div
    className="ant-skeleton"
  >
    <div
      className="ant-skeleton-content"
    >
      <h3
        className="ant-skeleton-title"
        style={
          Object {
            "width": "38%",
          }
        }
      />
      <ul
        className="ant-skeleton-paragraph"
      >
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": undefined,
            }
          }
        />
        <li
          style={
            Object {
              "width": "61%",
            }
          }
        />
      </ul>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
  >
    <div
      className="ant-form-item-label"
    >
      <label
        className=""
        htmlFor={null}
        onClick={[Function]}
        title="I Agree"
      >
        I Agree
      </label>
    </div>
    <div
      className="ant-form-item-control-wrapper"
    >
      <div
        className="ant-form-item-control has-success"
      >
        <span
          className="ant-form-item-children"
        >
          <label
            className="ant-checkbox-wrapper"
          >
            <span
              className="ant-checkbox"
              style={Object {}}
            >
              <input
                checked={false}
                className="ant-checkbox-input"
                name="agree"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                type="checkbox"
              />
              <span
                className="ant-checkbox-inner"
              />
            </span>
          </label>
        </span>
      </div>
    </div>
  </div>
  <button
    className="ant-btn ant-btn-primary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "marginRight": "10px",
      }
    }
    type="submit"
  >
    <span>
      Submit
    </span>
  </button>
</form>
`;
